packages/core/src/test/codewhisperer/util/zipUtil.test.ts

```diff
@@ -0,0 +1,153 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import assert from 'assert'
+import vscode from 'vscode'
+import sinon from 'sinon'
+import { join } from 'path'
+import { getTestWorkspaceFolder } from '../../../testInteg/integrationTestsUtilities'
+import { ZipUtil } from '../../../codewhisperer/util/zipUtil'
+import { SecurityScanType } from '../../../codewhisperer/models/constants'
+import * as CodeWhispererConstants from '../../../codewhisperer/models/constants'
+import { ToolkitError } from '../../../shared/errors'
+import fs from 'fs'
+import { fsCommon } from '../../../srcShared/fs'
+
+describe('zipUtil', function () {
```

Comment:
We are building a lot of scar tissue around adm-zip, but still haven't implemented a "streaming zip" approach. 

So we're building tech debt that gets us even further from our desired destination.

Reply:
I have the streaming zip approach ready locally, will raise another PR after this one